make on Mac breaks with a bad target for older sed

```
make generated_files DBG_MAKEFILE=1
Makefile:17: ***** starting makefile for goal(s) "generated_files"
Makefile:18: ***** Wed Jul 13 21:53:53 EDT 2016
Makefile.generated_files:93: ***** finding all *.go dirs
Makefile.generated_files:160: ***** finding all +k8s: tags
Makefile.generated_files:188: ***** finding all +k8s:deepcopy-gen tags
Makefile.generated_files:296: ***** finding all +k8s:conversion-gen tags
mkdir -p .make/_output/bin;                                        \
	(echo -n "_output/bin/conversion-gen: ";                        \
	 DIRECT=$(go list -e -f '{{.Dir}} {{.Dir}}/*.go' ./cmd/libs/go2idl/conversion-gen);                \
	 INDIRECT=$(go list -e   -f '{{range .Deps}}{{.}}{{"\n"}}{{end}}' ./cmd/libs/go2idl/conversion-gen | grep "^k8s.io/kubernetes" | xargs go list -e -f '{{.Dir}} {{.Dir}}/*.go');  \
	 echo $DIRECT $INDIRECT | sed 's/ / \\\n\t/g';       \
	) | sed "s|$(pwd -P)/||" > .make/_output/bin/conversion-gen.mk.tmp;                    \
	cmp -s .make/_output/bin/conversion-gen.mk.tmp .make/_output/bin/conversion-gen.mk || cat .make/_output/bin/conversion-gen.mk.tmp > .make/_output/bin/conversion-gen.mk && rm -f .make/_output/bin/conversion-gen.mk.tmp
mkdir -p .make/_output/bin;                                        \
	(echo -n "_output/bin/deepcopy-gen: ";                          \
	 DIRECT=$(go list -e -f '{{.Dir}} {{.Dir}}/*.go' ./cmd/libs/go2idl/deepcopy-gen);                  \
	 INDIRECT=$(go list -e   -f '{{range .Deps}}{{.}}{{"\n"}}{{end}}' ./cmd/libs/go2idl/deepcopy-gen | grep "^k8s.io/kubernetes" | xargs go list -e -f '{{.Dir}} {{.Dir}}/*.go');  \
	 echo $DIRECT $INDIRECT | sed 's/ / \\\n\t/g';       \
	) | sed "s|$(pwd -P)/||" > .make/_output/bin/deepcopy-gen.mk.tmp;                    \
	cmp -s .make/_output/bin/deepcopy-gen.mk.tmp .make/_output/bin/deepcopy-gen.mk || cat .make/_output/bin/deepcopy-gen.mk.tmp > .make/_output/bin/deepcopy-gen.mk && rm -f .make/_output/bin/deepcopy-gen.mk.tmp
make: *** No rule to make target `\nt/Users/clayton/projects/go/src/k8s.io/kubernetes/cmd/libs/go2idl/deepcopy-gen/main.go', needed by `_output/bin/deepcopy-gen'.  Stop.
```

I don't know where the `\nt` at the beginning of the target comes from, but probably from assumptions about sed version that are wrong.

I had to update to gnu-sed 4.2.2 to build.

@smarterclayton Are you sure the `Makefile` worked? I pull the latest from master, then run the same command above, I got:
```
F0714 13:36:50.652387   21108 main.go:76] Error: Failed making a parser: unable to add directory "k8s.io/kubernetes/pkg/apis/batch/v2alpha1": unable to import "k8s.io/kubernetes/pkg/apis/batch/v2alpha1": /home/cuonglm/go/src/k8s.io/kubernetes/pkg/apis/batch/v2alpha1/zz_generated.deepcopy.go:1:1: expected 'package', found 'EOF'
make: *** [Makefile.generated_files:201: gen_deepcopy] Error 255
```

Make sure you are in your GOPATH in the correct spot for Kube

On Jul 14, 2016, at 2:47 AM, Cuong Manh Le <notifications@github.com> wrote:

@smarterclayton <https://github.com/smarterclayton> Are you sure the
Makefile worked? I pull the latest from master, then run the same command
above, I got:

F0714 13:36:50.652387   21108 main.go:76] Error: Failed making a
parser: unable to add directory
"k8s.io/kubernetes/pkg/apis/batch/v2alpha1": unable to import
"k8s.io/kubernetes/pkg/apis/batch/v2alpha1":
/home/cuonglm/go/src/k8s.io/kubernetes/pkg/apis/batch/v2alpha1/zz_generated.deepcopy.go:1:1:
expected 'package', found 'EOF'
make: *** [Makefile.generated_files:201: gen_deepcopy] Error 255

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/kubernetes/issues/28930#issuecomment-232574285>,
or mute the thread
<https://github.com/notifications/unsubscribe/ABG_p5AsRKDQWBiGIPmZJ0PVAR5OLsWwks5qVdtmgaJpZM4JMBD0>
.

If you get the "expected 'package', found 'EOF'" message it is usually
because something somewhere failed, and left an empty .go file.  `make
clean_generated`

On Wed, Jul 13, 2016 at 11:47 PM, Cuong Manh Le <notifications@github.com>
wrote:

> @smarterclayton <https://github.com/smarterclayton> Are you sure the
> Makefile worked? I pull the latest from master, then run the same command
> above, I got:
>
> F0714 13:36:50.652387   21108 main.go:76] Error: Failed making a parser: unable to add directory "k8s.io/kubernetes/pkg/apis/batch/v2alpha1": unable to import "k8s.io/kubernetes/pkg/apis/batch/v2alpha1": /home/cuonglm/go/src/k8s.io/kubernetes/pkg/apis/batch/v2alpha1/zz_generated.deepcopy.go:1:1: expected 'package', found 'EOF'
> make: *** [Makefile.generated_files:201: gen_deepcopy] Error 255
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/issues/28930#issuecomment-232574285>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFVgVApOvl-1pF4ExQNMPgS2r0N3MP08ks5qVduGgaJpZM4JMBD0>
> .
>

--- a/Makefile.generated_files
+++ b/Makefile.generated_files
@@ -200,7 +200,7 @@ DEEPCOPY_FILES := $(addsuffix /$(DEEPCOPY_FILENAME), $(DEEPCOPY_DIRS))
 gen_deepcopy: $(DEEPCOPY_FILES)
 	if [[ -f $(META_DIR)/$(DEEPCOPY_GEN).todo ]]; then                \
 	    $(DEEPCOPY_GEN)                                               \
-	        -i $$(cat $(META_DIR)/$(DEEPCOPY_GEN).todo | paste -sd,)  \
+	        -i $$(cat $(META_DIR)/$(DEEPCOPY_GEN).todo | paste -sd, -)  \
 	        --bounding-dirs $(PRJ_SRC_PATH)                           \
 	        -O $(DEEPCOPY_BASENAME);                                  \
 	fi
@@ -244,7 +244,7 @@ $(META_DIR)/$(DEEPCOPY_GEN).mk:
 	     ./cmd/libs/go2idl/deepcopy-gen                    \
 	     | grep "^$(PRJ_SRC_PATH)"                         \
 	     | xargs go list -e -f '{{.Dir}} {{.Dir}}/*.go');  \
-	 echo $$DIRECT $$INDIRECT | sed 's/ / \\\n\t/g';       \
+	echo $$DIRECT $$INDIRECT | sed 's/ / =,/g' | tr '=,' '\n\t'; \
 	) | sed "s|$$(pwd -P)/||" > $@.tmp;                    \
 	cmp -s $@.tmp $@ || cat $@.tmp > $@ && rm -f $@.tmp
 
@@ -310,7 +310,7 @@ CONVERSION_FILES := $(addsuffix /$(CONVERSION_FILENAME), $(CONVERSION_DIRS))
 gen_conversion: $(CONVERSION_FILES)
 	if [[ -f $(META_DIR)/$(CONVERSION_GEN).todo ]]; then                \
 	    $(CONVERSION_GEN)                                               \
-	        -i $$(cat $(META_DIR)/$(CONVERSION_GEN).todo | paste -sd,)  \
+	        -i $$(cat $(META_DIR)/$(CONVERSION_GEN).todo | paste -sd, -)  \
 	        -O $(CONVERSION_BASENAME);                                  \
 	fi
 
@@ -420,7 +420,7 @@ $(META_DIR)/$(CONVERSION_GEN).mk:
 	     ./cmd/libs/go2idl/conversion-gen                  \
 	     | grep "^$(PRJ_SRC_PATH)"                         \
 	     | xargs go list -e -f '{{.Dir}} {{.Dir}}/*.go');  \
-	 echo $$DIRECT $$INDIRECT | sed 's/ / \\\n\t/g';       \
+	echo $$DIRECT $$INDIRECT | sed 's/ / =,/g' | tr '=,' '\n\t'; \
 	) | sed "s|$$(pwd -P)/||" > $@.tmp;                    \
 	cmp -s $@.tmp $@ || cat $@.tmp > $@ && rm -f $@.tmp
 

